### PR TITLE
Feature/output files

### PR DESCRIPTION
**Pull request by CatarinaCarolina**
write clustering and network output files

### CODE REVIEW

File: big_scape/cli/cluster_cli.py [hunk 1]
```diff
@@ -84,5 +84,5 @@ def cluster(ctx, *args, **kwargs):
     init_logger_file(ctx.obj)
 
     # run BiG-SCAPE cluster
-    print(ctx.obj)
+    # print(ctx.obj)
```
Comment (adraismawur) [inline big_scape/cli/cluster_cli.py:87]: Can be deleted

File: big_scape/main.py [hunk 2]
```diff
@@ -339,10 +341,15 @@ def align_callback(tasks_done: int):
     # prepare output files
     legacy_prepare_output(run["output_dir"], pfam_info)
 
+    # write full network file
+    write_full_network_file(run, all_bgc_records)
+
     # prepare output files per cutoff
     for cutoff in run["gcf_cutoffs"]:
-        # TODO: update to use records and not gbk regions
+        # TODO: update to use records and not gbk regions?
```
Comment (adraismawur) [inline big_scape/main.py:349]: TODO can be removed, I think

File: big_scape/output/legacy_output.py [hunk 3]
```diff
@@ -841,6 +843,9 @@ def generate_bs_families_alignment(
         # collects records within this GCF
         family_records = [records[bgc_num] for bgc_num in family_members]
         family_member_db_ids = [rec._db_id for rec in family_records]
+        if family_db_id not in family_member_db_ids:
+            continue
+        # TODO: get rid of this once bug in fam assignment fixed
```
Comment (adraismawur) [inline big_scape/output/legacy_output.py:848]: this should be in dev now, let me know if it's not

File: big_scape/output/legacy_output.py [hunk 4]
```diff
@@ -1146,8 +1154,452 @@ def legacy_generate_bin_output(
...
+
+            # for record B
+            select_statment = select(
+                bgc_record_table.c.gbk_id,
+                bgc_record_table.c.record_number,
+                bgc_record_table.c.record_type,
+            ).where(bgc_record_table.c.id == region_b_id)
+            gbk_id_b, record_number_b, record_type_b = DB.execute(
+                select_statment
+            ).fetchone()
```
Comment (adraismawur) [inline big_scape/output/legacy_output.py:1414]: challenge if you want it: there is a way to join these two select statements as one.

I think if we want to optimize it further, there should also be a way to cut out the middleman of passing existing_distances.

both of these can be TODO items as chores. they may be needed for very large datasets.

File: test/comparison/test_binning.py [hunk 5]
```diff
@@ -37,6 +37,11 @@ def create_mock_gbk(i, source_type: bs_enums.SOURCE_TYPE) -> GBK:
     cds.strand = 1
     gbk.genes.append(cds)
     gbk.region = Region(gbk, 1, 0, 100, False, "test")
+    gbk.metadata = {
+        "organism": "banana",
+        "taxonomy": "bananus;fruticus",
+        "description": "you can eat it",
```
Comment (adraismawur) [inline test/comparison/test_binning.py:43]: very true

**Review by adraismawur — CHANGES_REQUESTED** [review]
looks good to me. few changes

File: big_scape/main.py (hunk 2 above)
Comment (CatarinaCarolina) [reply big_scape/main.py:349]: legacy_prepare_cutoff_output()  still uses gbks, not records?

File: big_scape/output/legacy_output.py (hunk 3 above)
Comment (CatarinaCarolina) [reply big_scape/output/legacy_output.py:848]: I dont think so? maybe i'm missing it somehow

File: big_scape/output/legacy_output.py (hunk 4 above)
Comment (CatarinaCarolina) [reply big_scape/output/legacy_output.py:1414]: added as TODOs for now

**Pull request merged**